shell/platform/windows/accessibility_alert.cc

```diff
@@ -0,0 +1,166 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/accessibility_alert.h"
+
+#include "flutter/shell/platform/windows/accessibility_root_node.h"
+
+namespace flutter {
+
+AccessibilityAlert::AccessibilityAlert() : text_(L""), parent_(nullptr) {}
+
+IFACEMETHODIMP AccessibilityAlert::accHitTest(LONG screen_physical_pixel_x,
+                                              LONG screen_physical_pixel_y,
+                                              VARIANT* child) {
+  child->vt = VT_EMPTY;
+  return S_FALSE;
+}
+
+// Performs the object's default action.
+IFACEMETHODIMP AccessibilityAlert::accDoDefaultAction(VARIANT var_id) {
+  return E_FAIL;
+}
+
+// Retrieves the specified object's current screen location.
+IFACEMETHODIMP AccessibilityAlert::accLocation(LONG* physical_pixel_left,
+                                               LONG* physical_pixel_top,
+                                               LONG* width,
+                                               LONG* height,
+                                               VARIANT var_id) {
+  return E_NOTIMPL;
+}
+
+// Traverses to another UI element and retrieves the object.
+IFACEMETHODIMP AccessibilityAlert::accNavigate(LONG nav_dir,
+                                               VARIANT start,
+                                               VARIANT* end) {
+  end->vt = VT_EMPTY;
+  return E_NOTIMPL;
+}
+
+// Retrieves an IDispatch interface pointer for the specified child.
+IFACEMETHODIMP AccessibilityAlert::get_accChild(VARIANT var_child,
+                                                IDispatch** disp_child) {
+  if (V_VT(&var_child) == VT_I4 && V_I4(&var_child) == CHILDID_SELF) {
+    *disp_child = this;
+    AddRef();
+    return S_OK;
+  }
+  *disp_child = nullptr;
+  return E_FAIL;
+}
+
+// Retrieves the number of accessible children.
+IFACEMETHODIMP AccessibilityAlert::get_accChildCount(LONG* child_count) {
+  *child_count = 0;
+  return S_OK;
+}
+
+// Retrieves a string that describes the object's default action.
+IFACEMETHODIMP AccessibilityAlert::get_accDefaultAction(VARIANT var_id,
+                                                        BSTR* default_action) {
+  *default_action = nullptr;
+  return E_NOTIMPL;
+}
+
+// Retrieves the tooltip description.
+IFACEMETHODIMP AccessibilityAlert::get_accDescription(VARIANT var_id,
+                                                      BSTR* desc) {
+  *desc = SysAllocString(text_.c_str());
```

Comment:
Who deallocates the `desc` memory? Is this method called more than once? Will this allocate each time the method is called and leak the previous invocation's memory?

Reply:
The current implementations in the ax library allocate these properties similarly: https://github.com/flutter/engine/blob/main/third_party/accessibility/ax/platform/ax_platform_node_win.cc#L1056 and the example on Microsoft's page on this method does not outright state but seems to strongly imply that the caller is responsible for the memory: https://learn.microsoft.com/en-us/windows/win32/api/oleacc/nf-oleacc-iaccessible-get_accdescription